### ISSUE TITLE
switching styleURLs with vector tiles crashes in iOS, in native code

### ISSUE THREAD

**Issue by savv** [nitaliano/react-native-mapbox-gl#981]
Switching between two styles crashes the app in iOS, under specific conditions:
* the shape is within viewport
* in one of the styles, the shape is a Fill Layer, and in the other one it's a Line Layer

To replicate, in the [ShowMap.js](https://github.com/mapbox/react-native-mapbox-gl/blob/master/example/src/components/ShowMap.js) example, replace `this._mapOptions` with:
```
    this._mapOptions = [
      { label: 'fill', data: 'mapbox://styles/savvopoulos/cjcoycls73sxp2rp4co4qnboc', },
      { label: 'line', data: 'mapbox://styles/savvopoulos/cjcoy8i9t3ryg2rotek74nf2s' },
      { label: 'fill-copy', data: 'mapbox://styles/savvopoulos/cjcoyqkak3ogz2snset60eis6' }];
```

and replace: `userTrackingMode={MapboxGL.UserTrackingModes.Follow}`
with: `centerCoordinate={[8.52978140115738, 47.39449276792997]}`.
The issue reproduces only if the shape is within the viewport.

Launch the example app, go to "Show Map" and switch between fill and line; the app should crash with:
```
Thread 1: EXC_BAD_ACCESS (code=EXC_I386_GPFLT)
```

**Comment by savv** [comment]
Also note that starting out with line works, and that then switching to fill crashes. So nothing to do with the style itself, but the actual switching.

**Comment by savv** [comment]
As a workaround, I used a transparent fill layer, and used the 1px stroke to achieve a similar effect.

It would still be cool to see this fixed, of course :-)

**Comment by hampelm** [comment]
I'm seeing the same crash (`Thread 1: EXC_BAD_ACCESS (code=EXC_I386_GPFLT)`) switching from circle to polygon. (`6.0.3-rc1`, react-native `0.52.1`)

I see a similar-sounding issue in the native library's issues: https://github.com/mapbox/mapbox-gl-native/issues/11078

**Comment by hampelm** [comment]
Just wanted to bump this, since it's blocking some features we are hoping to roll out. Does this look like an upstream issue with `mapbox-gl-native`? Happy to put together a small sample repo if that's helpful.

**Comment by hampelm** [comment]
In case it's helpful for anyone else, the crash I was encountering happened when I switched between two layer types with the same source and id. The solution was to change the layer id.

Crashes when you switch between `a` & `!a`: 
```
if (a) {
  return <MapboxGL.CircleLayer id="mylayer" sourceLayerID="mysource" />
} else {
  return <MapboxGL.FillLayer id="mylayer" sourceLayerID="mysource" />
}
```

Fixed by using two different IDs: 

```
if (a) {
  return <MapboxGL.CircleLayer id="myCircleLayer" sourceLayerID="mysource" />
} else {
  return <MapboxGL.FillLayer id="myFillLayer" sourceLayerID="mysource" />
}
```

**Comment by nitaliano** [comment]
We made a fix for this recently and it is in `6.1.0`, this has been tested by someone else and they verified that it fixes there issue. I'm going to close this out but if this is still happening please reopen maybe we missed an edge case
